sdk/storage/azure-storage-blob/tests/test_blob_client.py

```diff
@@ -3,78 +3,94 @@
 from azure.core.credentials import AzureSasCredential
 from azure.core.exceptions import AzureError
 from azure.storage.blob import (
-    VERSION,
+    AccountSasPermissions,
+    BlobClient,
     BlobServiceClient,
     ContainerClient,
-    BlobClient,
+    generate_account_sas,
+    ResourceTypes,
+    VERSION,
 )
-from devtools_testutils import ResourceGroupPreparer, StorageAccountPreparer
+
+from devtools_testutils import recorded_by_proxy
+from devtools_testutils.storage import StorageRecordedTestCase
 from settings.testcase import BlobPreparer
-from devtools_testutils.storage import StorageTestCase
 
-# ------------------------------------------------------------------------------
 SERVICES = {
     BlobServiceClient: 'blob',
     ContainerClient: 'blob',
     BlobClient: 'blob',
 }
-
 _CONNECTION_ENDPOINTS = {'blob': 'BlobEndpoint'}
-
 _CONNECTION_ENDPOINTS_SECONDARY = {'blob': 'BlobSecondaryEndpoint'}
 
 
-class StorageClientTest(StorageTestCase):
-    def setUp(self):
-        super(StorageClientTest, self).setUp()
-        self.sas_token = self.generate_sas_token()
-        self.token_credential = self.generate_oauth_token()
+class TestStorageClient(StorageRecordedTestCase):
 
     # --Helpers-----------------------------------------------------------------
     def validate_standard_account_endpoints(self, service, url_type, name, storage_account_key):
-        self.assertIsNotNone(service)
-        self.assertEqual(service.account_name, name)
-        self.assertEqual(service.credential.account_name, name)
-        self.assertEqual(service.credential.account_key, storage_account_key)
-        self.assertTrue('{}.{}.core.windows.net'.format(name, url_type) in service.url)
-        self.assertTrue('{}-secondary.{}.core.windows.net'.format(name, url_type) in service.secondary_endpoint)
+        assert service is not None
+        assert service.account_name == name
+        assert service.credential.account_name == name
+        assert service.credential.account_key == storage_account_key
+        assert '{}.{}.core.windows.net'.format(name, url_type) in service.url
+        assert '{}-secondary.{}.core.windows.net'.format(name, url_type) in service.secondary_endpoint
+
+    def generate_fake_sas_token(self):
+        fake_key = "a" * 30 + "b" * 30
+
+        return "?" + generate_account_sas(
+            account_name="test",  # name of the storage account
+            account_key=fake_key,  # key for the storage account
+            resource_types=ResourceTypes(object=True),
+            permission=AccountSasPermissions(read=True, list=True),
+            start=datetime.now() - timedelta(hours=24),
+            expiry=datetime.now() + timedelta(days=8),
+        )
 
     # --Direct Parameters Test Cases --------------------------------------------
     @BlobPreparer()
-    def test_create_service_with_key(self, storage_account_name, storage_account_key):
-        # Arrange
+    def test_create_service_with_key(self, **kwargs):
```

Comment:
Are these live-only? I thought these would need a `@recorded_by_proxy` 👀

Reply:
Any test here that does not have `recorded_by_proxy` is because that test makes no network requests, so it doesn't need to be recorded. These files have quite a few instances of that where they are just testing client construction and such. They are more like unit tests.